tap_shopify/__init__.py

```diff
@@ -82,9 +95,13 @@ def get_discovery_metadata(stream, schema):
     mdata = metadata.write(mdata, (), 'table-key-properties', stream.key_properties)
     mdata = metadata.write(mdata, (), 'forced-replication-method', stream.replication_method)
 
-    if stream.replication_key:
+    if stream.replication_method=="INCREMENTAL" and stream.replication_key:
         mdata = metadata.write(mdata, (), 'valid-replication-keys', [stream.replication_key])
 
+    parent_tap_stream_id = getattr(stream, "parent", None)
+    if parent_tap_stream_id:
+        mdata = metadata.write(mdata, (), 'parent-tap-stream-id', parent_tap_stream_id)
+
```

Comment:
I am not sure if you have parent define in the child stream objects, please reverify.